### DISCUSSION TITLE
Userspace block device on top of multiple files

### DISCUSSION THREAD

**Discussion by OlegVladimirov** [ming1/ubdsrv#68] [Ideas]
Hi there! Is it possible to rise up ublksrv on top of multiple files, so than storage was chunked? Couple examples of such projects:

- https://github.com/tspoolst/fuse-blockfs
- https://github.com/iassasin/chunkbd

**Comment by ming1** [comment]
Yes, it is pretty easy to setup one target in this way, such as raid1, raid0,  raid10, or more complicated raid.

And raid is just one example, you can build any shape of target which is over
multiple files by implementing ->handle_io_async() callback.
